hooks/publish.py

```diff
@@ -182,14 +181,106 @@ def settings_schema(self):
                     "Dictionary of template_key/sg_field pairs to populate on "
                     "the PublishedFile entity."
                 )
-            }
+            },
+            "template_properties": {
+                "type": "list",
+                "values": {
+                    "type": "str",
+                },
+                "allows_empty": True,
+                "default_value": ["publish_name_template",
+                                  "publish_path_template",
+                                  "publish_symlink_template",
+                                  "publish_linked_entity_name_template"],
+                "description": "List of keys on the item settings that represent a template."
+            },
+            "submit_to_farm": {
+                "type": "bool",
+                "allows_empty": True,
+                "default_value": False,
+                "description": "Submit to farm!"
+            },
+            "unsupported_setting": {
+                "type": "list",
+                "allows_empty": True,
+                "default_value": ["bla1", "bla2"],
+                "description": "List Setting!"
+            },
+        }
+        schema["Settings To Display"] = {
```

Comment:
`Settings To Display` should just be a dictionary of setting name : initialization strategy, which is uniquely defined per setting. We should not be trying to control other parts of the dialog from here.